src/tree.rs

```diff
@@ -1760,21 +1716,13 @@ pub enum MergeState<'t> {
         local_node: Node<'t>,
         remote_node: Node<'t>,
     },
-}
 
-/// The reason for uploading or reuploading a merged descendant.
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
-pub enum UploadReason {
```

Comment:
The `UploadReason` thing ended up being unused, we're back to Booleans for `should_apply_item`, `should_apply_structure`, and `should_upload`.